pkg/auth/requests/service_account.go

```diff
@@ -0,0 +1,160 @@
+		secretLister:     scaledContext.Core.Secrets("").Controller().Lister(),
+		restConfigGetter: restConfigGetter,
+		authClientCreator: func(clusterID string) (kubernetes.Interface, error) {
+			return scaledContext.Wrangler.MultiClusterManager.K8sClient(clusterID)
+		},
+		clusterProxyConfigsGetter: scaledContext.Wrangler.Mgmt.ClusterProxyConfig().Cache(),
+	}
+}
+
+// Authenticate the request using the downstream service account's JWT.
+func (t *ServiceAccountAuth) Authenticate(req *http.Request) (user.Info, bool, error) {
+	// Basic checks to see if we even need to proceed
+	info, hasAuth := request.UserFrom(req.Context())
+	if info.GetName() != "system:cattle:error" {
+		return info, hasAuth, nil
+	}
+	clusterID := mux.Vars(req)["clusterID"]
+	if clusterID == "" {
+		return info, hasAuth, fmt.Errorf("no clusterID found in request")
+	}
+	// Check the cluster setting value to determine whether we will continue the auth process
+	settings, err := t.clusterProxyConfigsGetter.List(clusterID, labels.NewSelector())
+	if err != nil {
+		logrus.Debugf("rejecting downstream proxy request for %s, unable to fetch ClusterProxySettings object for cluster. %v", req.URL.Path, err)
+		return info, false, nil
+	}
+	if settings == nil {
+		logrus.Debugf("rejecting downstream proxy request for %s, no ClusterProxySettings object exists for cluster", req.URL.Path)
+		return info, false, nil
+	}
+	if len(settings) > 1 {
+		logrus.Errorf("multiple clusterproxyconfigs found for cluster, which is a misconfiguration, feature is disabled")
+		return info, false, nil
+	}
+
+	if !settings[0].Enabled {
+		logrus.Debugf("rejecting downstream proxy request for %s, current setting is enabled: %v", req.URL.Path, settings[0].Enabled)
+		return info, false, nil
+	}
+
+	// See if the token is a JWT.
+	rawToken := tokens.GetTokenAuthFromRequest(req)
+
+	jwtParser := jwtv4.NewParser(jwtv4.WithoutClaimsValidation())
+	claims := jwtv4.RegisteredClaims{}
+	// Using ParseUnverified is deliberate here to look at the basic info in the token.
+	// Later on, we do a real TokenReview against the downstream cluster to actually verify the JWT.
+	_, _, err = jwtParser.ParseUnverified(rawToken, &claims)
+	if err != nil {
+		logrus.Debug("saauth: error parsing JWT")
+		return info, false, err
+	}
+
+	if !strings.HasPrefix(claims.Subject, serviceaccount.ServiceAccountUsernamePrefix) {
+		logrus.Debugf("saauth: JWT sub is not a service account: %v", err)
+		return info, false, nil
+	}
+
+	if isTokenExpired(claims) {
+		logrus.Debugf("saauth: Service Account JWT is expired. Expiration time was: %v", claims.ExpiresAt)
+		return info, false, nil
+	}
+
+	// Get a client for the downstream cluster.
+	downstreamAuthClient, err := t.authClientCreator(clusterID)
+	if err != nil {
+		logrus.Errorf("saauth: failed to fetch downstream kubeconfig: %v", err)
+		return info, false, fmt.Errorf("failed to get downstream auth client when validating token for downstream cluster: %s", clusterID)
```

Comment:
Why do we return the error here, and  the error is not returned when the token is expired or token review couldn't be created? You mentioned [here](https://github.com/rancher/rancher/pull/43147/files#r1462316675) that we don't want to prevent other authenticators from being attempted. However,  based on my [findings](https://github.com/rancher/rancher/pull/43147/files#r1512549671) I believe that no other authenticator would be attempted after the SA token fails authentication. 
Unless we expect to add more authenticators in the future I would suggest to also return error [here](https://github.com/rancher/rancher/pull/43147/files#r1459174919), if there is any error parsing the jwt and if the token has expired.

Reply:
I don't know if I "expect" to add more authenticators in the future, but I also didn't expect to add this one.  
In order to be more consistent, how about we also don't return the error in this case?

Reply:
> how about we also don't return the error in this case?
that sounds good to me